File: 7-kyu/mean-means/javascript/solution.js
const geo_mean = (nums, arith_mean) => {
  const missingNumber = arith_mean * (nums.length + 1) - nums.reduce((acc, nxt) => acc + nxt, 0);
  nums.push(missingNumber);
  const geomMean = nums.reduce((acc, nxt) => acc * nxt, 1) ** (1 / nums.length);
  return geomMean;
}

File: 7-kyu/mean-means/python/solution.py
from functools import reduce
from operator import mul

def geo_mean(nums, arith_mean):
    arith_total = arith_mean * (len(nums) + 1)
    list_total = sum(nums)
    nums.append(arith_total - list_total)
    return reduce(mul, nums) ** (1 / len(nums))
